Documentation/Concepts/events.md

```diff
@@ -0,0 +1,133 @@
+        Generation int
+    }
+    Public bool
+}
+```
+
+For the whole structure of an event as defined in the gRPC, please check [Contracts](https://github.com/dolittle/Contracts/tree/master/Source/Runtime/Events).
+
+### Content
+This is the object the user commits.
+
+### EventSourceId
+`EventSourceId` represents the source of the event like a "primary key" in a traditional database. By default, [partitioned event handlers]({{< ref "event_handlers_and_filters#event-handlers" >}}) use it as the `PartitionId`.
+
+### EventType
+An `EventType` is a combination of an `EventTypeId` to uniquely identify the type of event it is and the event types `Generation`.
+This decouples the event from a programming language and enables the renaming of events as the domain language evolves.
+Simply put, event types are a wrapper for the actual type of your event. They mask the underlying language-specific type implementation and conventions.
+
+The Runtime doesn't know or care about the event's content, properties, or type (in its respective programming language). The Runtime saves the event to the event store and then calls the respective [EventHandlers & Filters]({{< ref "event_handlers_and_filters" >}}). For this event to be serialized to JSON and then deserialized back to a type that the client's filters and event handlers understand, an event type is required.
+
+This diagram shows us a simplified view of committing a single event with the type of `DishPrepared`. The Runtime receives the event, and sends it back to us to be handled. Without the event type, the SDK wouldn't know how to deserialize the JSON message coming from the Runtime.
+
+![Flow of committing an event type](/images/concepts/eventtype.png)
+
+Event types are also important when wanting to deserialize events coming from other microservices. As the other microservice could be written in a completely different programming language, event types provide a level of abstraction for deserializing the events.
+
+{{< alert title="Why not use class/type names instead of GUIDs?" color="info" >}}
+When consuming events from other microservices it's important to remember that they name things according to their own domain and conventions.
+
+As an extreme example, a microservice could have an event with a type `CustomerRegistered`. But in another microservice in a different domain, written in a different language, this event type could be called `user_added`.
+
+GUIDs also solve the problem of having duplicate names, it's not hard to imagine having to have multiple events with the type of `CustomerRegisterd` in your code coming from different microservices.
+{{< /alert >}}
+
+#### Generations
+As the code changes, the event type is also bound to change at some point. These iterations on the same event type are called _generations_. Whenever you add or change a property in an event you should increase this number. This way the filters and handlers can handle a specific generation of an event.
```

Comment:
I would say the structure of the event, rather than the evnet type
```suggestion
As the code changes, the structures and contents of your events are also bound to change at some point. In most scenarios you will see that you have the need to add more information to events. These iterations on the same event type are called _generations_. Whenever you add or change a property in an event you should increase this number. This way the filters and handlers can handle a specific generation of an event.
```